feat: enhance the "Awards" page ui

biru-codeastromer opened the pull request:
#### Fixes #7886 

### Description : -

- This PR updates the Awards page to improve its visual appeal while ensuring consistency with the Jenkins.io design language. 

### Changes : 

- Made the layout more similar to how it works around with the sig page
- This makes the pages more synchronized with our website


### State of the page after changes :

<img width="1512" alt="Screenshot 2025-02-22 at 11 28 48 AM" src="https://github.com/user-attachments/assets/4c07c268-dea2-486f-abeb-f0554487b6fb" />
<img width="1512" alt="Screenshot 2025-02-22 at 11 28 33 AM" src="https://github.com/user-attachments/assets/15f5cb1b-5143-4c20-bf29-3c370d4818d7" />


### Testing : 

- I ran `make` and `make run` to confirm website running .


Please let me know for improvements.

krisstern commented on content/awards.html.haml:
```diff
@@ -3,19 +3,22 @@ layout: simplepage
+            = award['quote']
+    - if award['image']
+      .award-image.text-center
+        %img{:src => expand_link('images/' + award['image']), :alt => award['title'], :class => 'img-fluid rounded'}
```
```suggestion
        %img{:src => expand_link('images/' + award['image']), :alt => award['title'], :class => 'img-fluid rounded'}
        
```

biru-codeastromer commented:
Done !

krisstern commented on content/awards.html.haml:
```diff
@@ -3,19 +3,23 @@ layout: simplepage
+    - if award['image']
+      .award-image.text-center
+        %img{:src => expand_link('images/' + award['image']), :alt => award['title'], :class => 'img-fluid rounded'}
+        
```
```suggestion

```

krisstern replied:
No extra spaces, but just a blank line

biru-codeastromer replied:
Done . Thanks  !

krisstern commented on content/awards.html.haml:
```diff
@@ -3,19 +3,22 @@ layout: simplepage
+            = award['quote']
+    - if award['image']
+      .award-image.text-center
+        %img{:src => expand_link('images/' + award['image']), :alt => award['title'], :class => 'img-fluid rounded'}
```
We want the blank line still, but without any extra spaces.

biru-codeastromer replied:
Done  !  Let me know if the blank line execution is proper now  .

biru-codeastromer replied:
Thanks .

krisstern commented:
@zbynek Would you like to give this a review?

timja commented:
cc @janfaracik

janfaracik commented on content/css/jenkins.css:
```diff
@@ -1550,3 +1550,180 @@ ion-icon.report {
   font-weight: bold;
 }
+
+/* Awards Page styling */
```
Hey.

This would be better in its own SCSS file (see `content/stylesheets/pages/_blog.scss`) this will make it easier to update in the future.

biru-codeastromer replied:
Sure thanks for the reference !

biru-codeastromer replied:
Done !

janfaracik commented on content/css/jenkins.css:
```diff
@@ -1550,3 +1550,180 @@ ion-icon.report {
+}
+
+.award-card:hover {
+  transform: translateY(-5px);
```
Not really sure about these animations, feels a bit odd that the card jumps up on hover, and the image has its own separate animation too.

biru-codeastromer replied:
Thanks for your feedback on this I would  avoid this and remove it.

biru-codeastromer replied:
Done !

janfaracik commented on content/css/jenkins.css:
```diff
@@ -1550,3 +1550,180 @@ ion-icon.report {
+
+/* Awards Page styling */
+:root {
+  --page-title-color: #0D1B2A;
```
We already have a bunch of CSS variables that can be used rather than defining new ones here.

biru-codeastromer replied:
I will be updating my changes in those and finally update it here after I have made my changes. Thank you for your guidance !

biru-codeastromer commented:
I have implemented the suggestions please let me know for further improvements if needed. Thanks!

biru-codeastromer commented:
I wanted to seek all reviewers thoughts on the potential addition of a gradient effect to the cards. Do you think this feature would align well with the overall design and coordinate effectively with the existing pages? Or do you feel it might not be a suitable feature to implement?

Looking forward to your suggestions and feedback!

zbynek commented on content/stylesheets/styles.scss:
```diff
@@ -3,3 +3,4 @@
 @import "core/_all";
 @import "components/_all";
 @import "pages/_all";
+@import "pages/awards";
```
pages/_all already includes your styles, please remove this import

zbynek commented on content/stylesheets/pages/_awards.scss:
```diff
@@ -0,0 +1,160 @@
+    text-decoration: none;
+
+    &:hover {
+      background: #1565c0; /* Manually darkened color */
```
should this be the same for light and dark theme?

zbynek commented on content/stylesheets/pages/_awards.scss:
```diff
@@ -0,0 +1,160 @@
+/* Awards Page Styling */
+:root {
+  --page-title-color: #0D1B2A;
```
It would be great to reuse as many existing color variables as possible.

biru-codeastromer replied:
I wanted to know  the use of existing colour variables means those present primarily inside our repo css files right . Just wanted to have the confirmation. Thanks

zbynek replied:
Exactly, I meant the ones defined in https://github.com/jenkins-infra/jenkins.io/blob/master/content/stylesheets/abstracts/_theme.scss

biru-codeastromer replied:
> Exactly, I meant the ones defined in https://github.com/jenkins-infra/jenkins.io/blob/master/content/stylesheets/abstracts/_theme.scss

Thanks !

zbynek reviewed (changes requested):
Maybe the image could be centered in the right column and it would be nice to simplify the CSS a bit as suggested in other comments.

biru-codeastromer commented:
> Maybe the image could be centered in the right column and it would be nice to simplify the CSS a bit as suggested in other comments.

I have done that I will push the changes.

biru-codeastromer commented:
I have removed the award import from `styles.scss` and also centered the image to the right column as suggested .

I wanted to have the clarification of the use of existing css variables . Like primarily for what fields will it be needed ??
I will now look the variables available to use and change them too .

kmartens27 commented on content/awards.html.haml:
```diff
@@ -3,19 +3,28 @@ layout: simplepage
+  .container
+    .row
+      .col-md-12
+        %p.sub-heading Celebrating the milestones and achievements of Jenkins.
```
```suggestion
        %p.sub-heading Highlighting the milestones and achievements of Jenkins.
```
Just a small suggestion, not necessary to implement though.

kmartens27 reviewed (approved):
I think the award page looks nice with the updated visuals. Thanks very much @biru-codeastromer, once some more reviews are submitted and any feedback is actioned on, we can look towards merge!

biru-codeastromer commented:
> I think the award page looks nice with the updated visuals. Thanks very much @biru-codeastromer, once some more reviews are submitted and any feedback is actioned on, we can look towards merge!

Sure !

krisstern reviewed (commented):
For the mobile version I noticed some of the images are not properly centered, is it possible to update this via some CSS changes?

janfaracik commented:
<img width="1464" alt="image" src="https://github.com/user-attachments/assets/d915ef44-e66f-45f3-9427-716f461d27ea" />

It'd be great to see these elements aligned to the left and I feel like the cards are taking up quite a lot of space without much benefit. See https://deploy-preview-7888--jenkins-io-site-pr.netlify.app/sigs/ for example, it's quite a bit more compact but still gets the information across to the user. We could do away with a lot of custom variables too if we dropped the cards.

biru-codeastromer commented:
> For the mobile version I noticed some of the images are not properly centered, is it possible to update this via some CSS changes?


Thanks for the point out , I will fix it to have mobile versions better too .

biru-codeastromer commented:
> It'd be great to see these elements aligned to the left and I feel like the cards are taking up quite a lot of space without much benefit. See https://deploy-preview-7888--jenkins-io-site-pr.netlify.app/sigs/ for example, it's quite a bit more compact but still gets the information across to the user. We could do away with a lot of custom variables too if we dropped the cards.
> 
> Might be worth having a placeholder image too for awards which dont have an image so that the description is aligned consistently.



Thanks for your suggestions . I just had a look into the sigs page , TBH I found it really cool and aesthetic .
I will surely now try to implement a similar UI for this page as both of them does somehow seems to have a similar kind of layout .

Thanks for the reference again !

Will update here after I have implemented them .

biru-codeastromer commented:
Hello 👋,

I tried to implement the same kind of UI for award page to that of sig page to maintain consistency in our pages .
Also I have fixed the mobile version layout issue .

Let me know how it looks now the UI .
Please give me suggestions for improvement if any. 
Thanks !

biru-codeastromer commented:
I did want to know I have added a inclusion of placeholder image, but not have added one . So , what might be the best choice for the placeholder image.

biru-codeastromer commented:
<img width="1512" alt="Screenshot 2025-02-21 at 8 56 50 PM" src="https://github.com/user-attachments/assets/8b0a529d-2cb5-43a4-9a4c-c731646add89" />


The present state ...

krisstern commented:
The placeholder image is "broken" / not showing in the preview presently: 

<img width="1435" alt="Screenshot 2025-02-21 at 11 30 52 PM" src="https://github.com/user-attachments/assets/841a7e79-02cb-47b5-865c-17ee9ada3b13" />

krisstern commented:
Maybe you can just use the Jenkins butler as the placeholder image? This is not very professionally done if you intend to show a placeholder image but you are not providing any such placeholder image.

biru-codeastromer commented:
Let me fix it properly that placeholder image .

biru-codeastromer commented:
> Maybe you can just use the Jenkins butler as the placeholder image? This is not very professionally done if you intend to show a placeholder image but you are not providing any such placeholder image.



Hi , I have fixed the placeholder image issue . Let me how it goes now . Thanks !

krisstern commented on content/stylesheets/pages/_awards.scss:
```diff
@@ -0,0 +1,98 @@
+      }
+    }
+  }
+}
```
```suggestion
}

```

biru-codeastromer commented:
Done , I have applied the change .

kmartens27 commented:
I like the updated formatting for the page, it appears closer in style to the SIGs page that was recently updated.

krisstern commented on content/stylesheets/pages/_awards.scss:
```diff
@@ -0,0 +1,98 @@
+  .sub-heading {
+    font-size: 1.2rem;
+    color: var(--color--secondary);
+    text-align: center;
```
```suggestion
    text-align: start;
```

I am thinking maybe the sub-heading better be left-aligned?

krisstern commented:
> I like the updated formatting for the page, it appears closer in style to the SIGs page that was recently updated.

I echo that sentiment too. It looks a lot more in keeping with our overall SIG CSS styling standards now!

biru-codeastromer commented:
Done , I have applied .

biru-codeastromer commented:
<img width="1512" alt="Screenshot 2025-02-21 at 10 59 39 PM" src="https://github.com/user-attachments/assets/b3158ae4-4bc2-493b-af90-8f7628c6e6be" />


Do you believe a little bit of padding would be great between sub-heading and the heading

janfaracik commented:
<img width="286" alt="image" src="https://github.com/user-attachments/assets/b9f77414-0a19-4e25-b904-2b761e60d523" />

Hey, looking better - thanks! Would be good to get these aligned, the description ('Highlighting the milestones...') has a couple unnecessary wrapping divs that can be removed.

I don't think the italics are helping anything here - we could probably do away with the inset border too.

biru-codeastromer commented:
Sure , let me fix them and implement your suggestions .

biru-codeastromer commented:
<img width="1512" alt="Screenshot 2025-02-22 at 11 28 48 AM" src="https://github.com/user-attachments/assets/fe40910f-c594-437d-a1d2-65c4ed40dbda" />


-----


<img width="1512" alt="Screenshot 2025-02-22 at 11 28 33 AM" src="https://github.com/user-attachments/assets/3a91b93b-828b-4927-8ae6-bb5effc43bfc" />

biru-codeastromer commented:
Hello 👋, 
I have implemented the changes provided by Jan .
I have made few more adjustments to make the page more aligned with how the sig page works , now it has become more like replica to it . This synchronises our website pages .


Let me know if any improvements needed for more betterment .
Thanks

krisstern reviewed (approved):
This looks good to me now. Thanks for all the hard work @biru-codeastromer!

biru-codeastromer commented:
> This looks good to me now. Thanks for all the hard work @biru-codeastromer!

You are welcome 😊 ..

janfaracik commented:
That's looking great! Thanks. 

I'm on the move at the moment so just a few comments, would be good to have the image aligned to the top rather than center, and then removing the rounded corners of the image and quote.


The placeholder Jenkins logo is also quite low res - there'll be an SVG floating around the codebase if you could replace it with that :) 

Great job.

biru-codeastromer commented:
> That's looking great! Thanks.
> 
> I'm on the move at the moment so just a few comments, would be good to have the image aligned to the top rather than center, and then removing the rounded corners of the image and quote.
> 
> The placeholder Jenkins logo is also quite low res - there'll be an SVG floating around the codebase if you could replace it with that :)
> 
> Great job.





Hi , I have made  your suggestions . Let me how it goes now . Thanks !

zbynek commented on content/_layouts/simplepage.html.haml:
```diff
@@ -15,7 +15,7 @@ layout: default
     %div{:class => "app-!-display-flex", :style => "margin-bottom: 20px"}
       %a{:href => page.backlink.url, :class => "app-back-link"}
         = page.backlink.label
-  - unless page.notitle
```
Let's keep the `notitle` attribute.

biru-codeastromer replied:
when we keep this, there comes two titles with "awards" ; that's why I tried to implement that logic .

biru-codeastromer replied:
Should I use a combination of both logic ?? what you think about it??

biru-codeastromer commented:
I have retained back the `notitle` attribute again .

krisstern commented on content/_layouts/simplepage.html.haml:
```diff
@@ -15,7 +15,7 @@ layout: default
     %h1
       = page.title
-  = content
+  = content
```
```suggestion
  = content
  
```

krisstern commented on content/_layouts/simplepage.html.haml:
```diff
@@ -15,7 +15,7 @@ layout: default
       %a{:href => page.backlink.url, :class => "app-back-link"}
         = page.backlink.label
-  - unless page.notitle
+  - if page.notitle != false
```
```suggestion
  - unless page.notitle
```

krisstern commented on content/_layouts/simplepage.html.haml:
```diff
@@ -19,3 +19,4 @@ layout: default
     %h1
       = page.title
   = content
+  
```
```suggestion

```

krisstern replied:
My apologies, please remove the spaces

biru-codeastromer replied:
No problem!

biru-codeastromer commented:
<img width="1512" alt="Screenshot 2025-02-23 at 11 20 35 AM" src="https://github.com/user-attachments/assets/f5412adb-7880-4025-9fb3-736fe873bb2e" />


I wanted to tell if we remove that if logic from simplepage.html.haml file we do come up with two titles for which I used that logic in that page .

krisstern commented:
> I wanted to tell if we remove that if logic from simplepage.html.haml file we do come up with two titles for which I used that logic in that page.

I don't think you are supposed to change the logic of the "simplepage.html.haml", since other pages are relying on it so there may be unintended consequences. But we could update the logic in "content/awards.html.haml".

krisstern reviewed (commented):
Maybe make these changes?

biru-codeastromer commented:
> > I wanted to tell if we remove that if logic from simplepage.html.haml file we do come up with two titles for which I used that logic in that page.
> 
> I don't think you are supposed to change the logic of the "simplepage.html.haml", since other pages are relying on it so there may be unintended consequences. But we could update the logic in "content/awards.html.haml".



Got it .

biru-codeastromer commented:
> Maybe make these changes?

By applying , I ran into few issues in ci checks, let me fix that now and see how we can achieve it without affecting `simplepage.html.haml` file

biru-codeastromer commented:
Done ! ✅

We needed to make `notitle: True` in our "awards.html.haml" file to resolve our issue and have one title without affecting the "simplepage.html.haml" file .

biru-codeastromer commented:
<img width="1512" alt="Screenshot 2025-02-23 at 1 12 29 PM" src="https://github.com/user-attachments/assets/53331b46-3dab-4098-a4a8-4c2bb6ecf5fd" />

janfaracik commented:
Hey! Looking good, thank you :) Last things from me, gave the branch a pull and did some tidying up:

```
---
layout: simplepage
title: Awards
tight: true
---

%p.app-description Highlighting the milestones and achievements of Jenkins.

- site.awards.values.sort_by { |award| [award['year'], award['name']] }.reverse.each do |award|
  .award-item
    .award-item__icon
      - if award['image']
        %img{src: expand_link('images/' + award['image']), alt: award['title'], width: "96"}
      - else
        %img{src: expand_link('/images/logos/jenkins/jenkins.svg'), alt: 'Placeholder', width: "96"}
    .award-item__content
      %h2.award-title
        - if award['url']
          %a{href: award['url'], target: '_blank', class: 'text-decoration-none'}
            = award['title']
        - else
          = award['title']
      - if award['description']
        %p.award-description
          = award['description']
      - if award['quote']
        %blockquote.award-quote
          %p.mb-0
            = award['quote']
```

```
.award-item {
  display: grid;
  grid-template-columns: auto 1fr;
  gap: 1.5rem;
  align-items: start;
  margin-bottom: 2rem;

  &__icon {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 110px;
    height: 110px;
    overflow: hidden;
    background-color: var(--background);

    img {
      width: 100%;
      height: 100%;
      object-fit: contain;
      object-position: top;
    }
  }

  &__content {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;

    .award-title {
      font-size: 1.25rem;
      font-weight: 500;
      margin: 0;

      a {
        color: var(--blue);
        text-decoration: none;

        &:hover {
          text-decoration: underline;
        }
      }
    }

    .award-description {
      font-size: 0.95rem;
      color: var(--color);
      line-height: 1.6;
      margin: 0;
    }

    .award-quote {
      font-size: 1rem;
      font-style: italic;
      color: var(--color--secondary);
      padding-left: 1rem;
      margin: 1rem 0;
    }
  }
}

@media (max-width: 768px) {
  .award-item {
    gap: 1rem;

    &__icon {
      width: 80px;
      height: 80px;
    }

    &__content {
      .award-title {
        font-size: 1.1rem;
      }

      .award-description {
        font-size: 0.9rem;
      }
    }
  }
}
```

Removes quite a few lines of code that aren't needed, `simplepage` has a `tight` property you can use which makes the page width smaller - removing the need for the custom container. Let me know what you think.

biru-codeastromer commented:
Thanks a lot, @janfaracik! 🚀 I learned a lot about keeping UI changes simple, something Kris also mentioned to me earlier.

The removal of extra lines of code and the use of the tight property are perfect—I tested it, and everything worked flawlessly.

While using custom properties wrapped inside a container, I had to test the page multiple times (at 1200px, 1100px, 1050px) to ensure it matched the SIG page 😄. This feature made that process much easier.

Really appreciate it—thanks again!

janfaracik reviewed (approved):
Looking great, thanks!

krisstern commented on content/awards.html.haml:
```diff
@@ -1,21 +1,29 @@
+      - if award['quote']
+        %blockquote.award-quote
+          %p.mb-0
+            = award['quote']
```
```suggestion
            = award['quote']

```

biru-codeastromer commented:
> Looking great, thanks!

Welcome 🤗

krisstern commented on content/stylesheets/pages/_awards.scss:
```diff
@@ -0,0 +1,81 @@
+      }
+    }
+  }
+}
```
```suggestion
}

```

biru-codeastromer commented:
Thanks a lot Kris  for the reminder of the blank line , I always missed that in new push . I will definitely keep that in mind :).

biru-codeastromer commented:
Done ! Brought back the blank line . 
Thanks

krisstern commented:
Thanks for the PR @biru-codeastromer!

biru-codeastromer commented:
> Thanks for the PR @biru-codeastromer!

You are welcome ! 🤗
